nni/compression/pytorch/compressor.py

```diff
@@ -38,12 +39,36 @@ def __init__(self, model, config_list, optimizer=None):
         optimizer: pytorch optimizer
             optimizer used to train the model
         """
+        self.optimizer = optimizer
+        if optimizer is not None:
+            self.optimizer_cls = optimizer.__class__
+            self.optimizer_param_groups = optimizer.state_dict()['param_groups']
+            # don't save 'params' in a 'param_group' in 'param_groups'
+            [group.pop('params') for group in self.optimizer_param_groups]
+        if model is not None and config_list is not None:
+            self.reconfig(model, config_list)
+
+    def reconfig(self, model, config_list):
```

Comment:
maybe we can split this function to two functions. one is `reset_status`, the other is `reset_config`. There could be another function `reset` which contains the above two functions

Reply:
`reset_config` rewrap the `new_model` with `new_config`, and `reset_status` used to reset the median params or all other params?

Reply:
`reset_with_new_config` is for resetting the pruner, it invokes `reset_status` and `set_config`.

Reply:
split it.